server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueMailStore.java

```diff
@@ -26,38 +26,59 @@
 import java.util.concurrent.ConcurrentHashMap;
 
 import javax.inject.Inject;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
 
+import org.apache.james.blob.api.Store;
+import org.apache.james.blob.mail.MimeMessagePartsId;
 import org.apache.james.queue.rabbitmq.MailQueueName;
 import org.apache.james.queue.rabbitmq.view.cassandra.model.BucketedSlices.BucketId;
 import org.apache.james.queue.rabbitmq.view.cassandra.model.EnqueuedMail;
 import org.apache.james.queue.rabbitmq.view.cassandra.model.MailKey;
+import org.apache.james.util.CompletableFutureUtil;
 import org.apache.mailet.Mail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 class CassandraMailQueueMailStore {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraMailQueueMailStore.class);
+
     private final EnqueuedMailsDAO enqueuedMailsDao;
     private final BrowseStartDAO browseStartDao;
+    private final Store<MimeMessage, MimeMessagePartsId> mimeMessageStore;
     private final CassandraMailQueueViewConfiguration configuration;
     private final Clock clock;
     private final Set<MailQueueName> initialInserted;
 
     @Inject
     CassandraMailQueueMailStore(EnqueuedMailsDAO enqueuedMailsDao,
                                 BrowseStartDAO browseStartDao,
+                                Store<MimeMessage, MimeMessagePartsId> mimeMessageStore,
                                 CassandraMailQueueViewConfiguration configuration,
                                 Clock clock) {
         this.enqueuedMailsDao = enqueuedMailsDao;
         this.browseStartDao = browseStartDao;
+        this.mimeMessageStore = mimeMessageStore;
         this.configuration = configuration;
         this.clock = clock;
         this.initialInserted = ConcurrentHashMap.newKeySet();
     }
 
-    CompletableFuture<Void> storeMailInEnqueueTable(Mail mail, MailQueueName mailQueueName, Instant enqueuedTime) {
-        EnqueuedMail enqueuedMail = convertToEnqueuedMail(mail, mailQueueName, enqueuedTime);
+    CompletableFuture<Void> storeMail(Mail mail, MailQueueName mailQueueName, Instant enqueuedTime) {
+        return storeMimeMessage(mail)
```

Comment:
This is bad: **Don't we already store the blob as part of RabbitMQ enqueue operation?

Can't we rather pass the blobIds as part of the `storeMail` operation?